launcher/modplatform/import_ftb/PackHelpers.cpp

```diff
@@ -27,6 +27,37 @@
 
 namespace FTBImportAPP {
 
+QIcon loadFTBIcon(const QString& imagePath)
+{
+    // Map of type byte to image type string
+    static const QMap<quint8, QString> imageTypeMap = { { 0x00, "png" }, { 0x01, "jpg" }, { 0x02, "gif" }, { 0x03, "webp" } };
+
+    QFile file(imagePath);
+    if (!file.exists() || !file.open(QIODevice::ReadOnly)) {
+        return QIcon();
+    }
+
+    QByteArray rawData = file.readAll();
+    if (rawData.size() < 2) {
+        return QIcon();
+    }
+
+    quint8 typeByte = static_cast<quint8>(rawData.at(0));
+    QString imageType = imageTypeMap.value(typeByte, QString());
+    if (imageType.isEmpty()) {
+        return QIcon();
+    }
+
+    // Extract actual image data (excluding the first byte)
+    QByteArray imageData = rawData.mid(1);
+    QPixmap pixmap;
+    if (!pixmap.loadFromData(imageData, imageType.toUtf8().constData())) {
+        return QIcon();
+    }
+
+    return QIcon(pixmap);
```

Comment:
I would make quite a few changes here lol
I assume this works...
Best to test it though lol
```suggestion
    // Map of type byte to image type string
    static const QHash<char, QByteArray> imageTypeMap = { { 0x00, "png" }, { 0x01, "jpg" }, { 0x02, "gif" }, { 0x03, "webp" } };

    QFile file(imagePath);
    if (!file.exists() || !file.open(QIODevice::ReadOnly)) {
        return QIcon();
    }

    char type;
    if (file.getChar(&type)) {
        qDebug() << "Missing FTB image type header at" << imagePath;
        return QIcon();
    }

    if (!imageTypeMap.contains(type)) {
        qDebug().nospace().noquote() << "Don't recognize FTB image type 0x" << QString::number(type, 16);
        return QIcon();
    }
    
    QString imageType = imageTypeMap[type];

    // Extract actual image data beyond the first byte
    QImageReader reader(&file, imageType);
    auto pixmap = QPixmap::fromImageReader(&reader);
    // note: will this actually be null if an error occurs?
    // note: should this log anything?
    if (pixmap.isNull()) {
        return QIcon();
    }

    return QIcon(pixmap);
```